Support for boolean data types as metrics

miguelHx opened the pull request:
This is useful if your database uses true/false for state and want to make prometheus alerts based on that.
Before, booleans were not able to be parsed.  See issue #201

wrouesnel commented:
This looks like it has linter errors.

miguelHx commented:
Ok, will work on it right now.

miguelHx commented:
@wrouesnel how do I run the linter?

miguelHx commented:
@wrouesnel checks passed, ready to merge whenever you are ready.

miguelHx commented:
Would you recommend that I write some unit tests for my code?

miguelHx commented:
It seems that the functions dbToFloat64 and dbToString are not covered by tests.  Since this is the case, how would I increase coverage from coveralls?

miguelHx commented:
It doesn't look like the CI error was introduced by my change. @wrouesnel 
```
0.01s$ $HOME/gopath/bin/goveralls -coverprofile=cover.out -service=travis-ci
Error parsing coverage: open cover.out: no such file or directory
The command "$HOME/gopath/bin/goveralls -coverprofile=cover.out -service=travis-ci" exited with 1.
```

wrouesnel commented:
@miguelHx Generally that happens when the smoke-tests fail in some way. But I'll take a look!

miguelHx commented:
@wrouesnel I ran the command `go run mage.go -v all` on my local machine.  I did not get the linter warnings from the travis build.  Could you trigger another build with a fresh pull?  I'm thinking it's a false positive.

miguelHx commented:
@wrouesnel can I get some eyes on this?  I see some parsing errors in the build log but I don't see how that relates to my code updates.

I think this is what's preventing a successful build (near the top of the build log):
```
The command "./gh-assets-clone.sh" exited with 1.
```
I can't find that particular script.

Any help would be appreciated.

miguelHx commented:
Found the script, going to take a look.

miguelHx commented:
@wrouesnel I'm looking at the script:
```bash
#!/bin/bash
# Script to setup the assets clone of the repository using GIT_ASSETS_BRANCH and
# GIT_API_KEY.

[ ! -z "$GIT_ASSETS_BRANCH" ] || exit 1
[ ! -z "$GIT_API_KEY" ] || exit 1

setup_git() {
  git config --global user.email "travis@travis-ci.org" || exit 1
  git config --global user.name "Travis CI" || exit 1
}

# Constants
ASSETS_DIR=".assets-branch"

# Clone the assets branch with the correct credentials
git clone --single-branch -b "$GIT_ASSETS_BRANCH" \
    "https://${GIT_API_KEY}@github.com/${TRAVIS_REPO_SLUG}.git" "$ASSETS_DIR" || exit 1
```
Do I need to update something in my branch to make this work?  I'm a little lost lol.

miguelHx commented:
Added some print statements in the script so narrow down where the script is exiting.
The script seems to exit here:
```[ ! -z "$GIT_API_KEY" ] || exit 1```

miguelHx commented:
As per my issue #235, now the build passes after the small `gh-assets-clone.sh` change.

miguelHx commented:
triggering an external build to test out latest change

miguelHx commented:
@wrouesnel Thank you Will!  I think it's ready to merge to master now:)

miguelHx commented:
There are many commits in this PR, so if you could click the option to merge all of this as one commit, that would be great.

wrouesnel commented:
Ah right, it needs a rebase. I was meaning to click the merge button a while back. I've pushed a best guess at the rebase back to the PR, we'll see if it builds properly...

miguelHx commented:
@wrouesnel Okay, looks like it built properly.  Let me know if any concerns arise.

miguelHx commented:
@wrouesnel Is the rebase complete?

miguelHx commented:
@wrouesnel my apologies, I realize that I should do the rebase.  I haven't done a rebase before, so I'll do some research and I'll make it happen.

miguelHx commented:
squashed my commits and rebased onto the upstream master.  Build failed, I'm worried that my rebase may have had something to do with it :0, but could be a different issue

miguelHx commented:
Rebased.  I think it's ready to be merged :)

miguelHx commented:
Friendly reminder :) @wrouesnel
